### PR TITLE
config_tools: clean up the board folders

### PR DESCRIPTION
**Pull request by Kunhui-Li**
1. Remove apl-up2, apl-up2-n3350, whl-ipc-i5, apl-mrb, nuc6cayh board
   folders from the latest code base.
2. Copy tgl-rvp.xml to generic_board.xml.

Tracked-On: #6175
Signed-off-by: Kunhui-Li <kunhuix.li@intel.com>

### CODE REVIEW

**Review by gvancuts — CHANGES_REQUESTED** [review]
There is quite a bit of documentation to be cleaned as well with that. We still use some of these boards in some build instructions. Some of these boards are also mentionned in our generic doc (inc. HLD) and last but not least, we should remove those from the "Supported Hardware List" document too.

**Comment by dbkinder** [comment]
There are documents that reference some of these deleted XML files that need to be updated, for example:
```
getting-started/building-from-source.rst:* Build the ``SDC`` scenario on the ``nuc6cayh``:
getting-started/building-from-source.rst:    $ make all BOARD_FILE=$PWD/misc/vm_configs/xmls/board-xmls/nuc6cayh.xml \
getting-started/building-from-source.rst:    SCENARIO_FILE=$PWD/misc/vm_configs/xmls/config-xmls/nuc6cayh/sdc.xml
reference/hardware.rst:   https://www.intel.com/content/www/us/en/products/boards-kits/nuc/kits/nuc6cayh.html
```

File: doc/tutorials/rdt_configuration.rst [hunk 1]
```diff
@@ -257,10 +257,10 @@ Configure RDT for VM Using VM Configuration
 
    .. code-block:: none
 
-      $ make hypervisor BOARD=apl-up2 FIRMWARE=uefi
+      $ make hypervisor BOARD=tgl-rvp FIRMWARE=uefi
```
Comment (gvancuts) [inline doc/tutorials/rdt_configuration.rst:260]: The `FIRMWARE` parameter is no longer valid. We need to remove it and update this with an appropriate pair of `SCENARIO` and `BOARD` parameters

File: doc/tutorials/rdt_configuration.rst [hunk 2]
```diff
@@ -257,10 +257,10 @@ Configure RDT for VM Using VM Configuration
...
       ...
 
-      # these operations are done on UP2 board
+      # these operations are done on tgl-rvp board
```
Comment (gvancuts) [inline doc/tutorials/rdt_configuration.rst:263]: We need a deeper look at that doc, do we use an eMMC storage device for example on this `tgl-rvp` board?

The doc also talks about `acrn.efi` which we need to update.

File: doc/developer-guides/hld/hv-dev-passthrough.rst [hunk 3]
```diff
@@ -382,9 +382,8 @@ the current VM, otherwise, none of them should be assigned to the
 current VM. A device that violates the rule will be rejected to be
 passed-through. The checking logic is implemented in Device Model and not
 in scope of hypervisor.
-The platform GSI information is in devicemodel/hw/pci/platform_gsi_info.c
-for limited platform (currently, only APL MRB). For other platforms, the platform
-specific GSI information should be added to activate the checking of GSI sharing violation.
+The platform specific GSI information shall be filled in devicemodel/hw/pci/platform_gsi_info.c
```
Comment (gvancuts) [inline doc/developer-guides/hld/hv-dev-passthrough.rst:385]:
```suggestion
The platform specific GSI information shall be filled in :file:`devicemodel/hw/pci/platform_gsi_info.c`
```

Does that look right @dbkinder ?

**Review by gvancuts — CHANGES_REQUESTED** [review]
We need a deeper look at the RDT doc:
* Does TGL support RDT/CAT?
* If so, are all the `rdmsr` and `cpuid` command examples in the doc accurate and representative?

Thanks

File: doc/developer-guides/hld/hv-dev-passthrough.rst (hunk 3 above)
Comment (dbkinder) [reply doc/developer-guides/hld/hv-dev-passthrough.rst:393]: LGTM

File: doc/tutorials/rdt_configuration.rst (hunk 1 above)
Comment (dbkinder) [reply doc/tutorials/rdt_configuration.rst]: change to a board in the supported hardware document

File: doc/getting-started/building-from-source.rst [hunk 4]
```diff
@@ -179,12 +179,12 @@ the following to build the hypervisor, device model, and tools:
 
 You can also build ACRN with your customized scenario:
 
-* Build with your own scenario configuration on the ``nuc6cayh``, assuming the
+* Build with your own scenario configuration on the ``nuc11tnbi5``, assuming the
```
Comment (dbkinder) [inline doc/getting-started/building-from-source.rst:182]: we should only reference boards in the supported hardware document (this one's not there)

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/getting-started/building-from-source.rst (hunk 4 above)
Comment (NanlinXie) [reply doc/getting-started/building-from-source.rst:182]: We will update the TGL NUC platform later.

File: doc/tutorials/rdt_configuration.rst (hunk 2 above)
Comment (Kunhui-Li) [reply doc/tutorials/rdt_configuration.rst]: Updated.

File: doc/tutorials/rdt_configuration.rst (hunk 1 above)
Comment (Kunhui-Li) [reply doc/tutorials/rdt_configuration.rst]: Updated.

**Comment by dbkinder** [comment]
@Kunhui-Li are you going to update the board names in documentation that refer to a board name you've deleted? @NanlinXie

**Comment by Kunhui-Li** [comment]
We remove the detailed command list and replace with general comments with link to GSG in rdt document , and it refer to whl-ipc-i7 board whose folder don't be removed in the PR. (GSG Link: https://projectacrn.github.io/latest/getting-started/rt_industry_ubuntu.html#build-the-acrn-hypervisor-on-ubuntu)

**Comment by gvancuts** [comment]
I think @dbkinder is referring to the list of boards in `reference/hardware.rst` (Supported Hadware). That list needs to be cleaned accordingly.

**Comment by NanlinXie** [comment]
@dbkinder , could we merge this one?

**Review by NanlinXie — APPROVED** [review]

**Pull request merged**